### PR TITLE
[GH-431] Add unit tests for ./store/sqlstore/workspaces.go

### PR DESCRIPTION
**Pull request by yedamao**
#### Summary

Add unit tests for ./store/sqlstore/workspaces.go

#### Ticket Link

Fixes https://github.com/mattermost/focalboard/issues/431

### CODE REVIEW

**Comment by yedamao** [comment]
insert workspace by store.UpsertWorkspaceSettings will failed, because signup_token NOT NULL constraint.
I'm not sure it's an expected error or a bug.

File: server/services/store/storetests/workspaces.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	t.Run("Insert and update workspace with signup token", func(t *testing.T) {
+		workspaceID := "0"
+		workspace := &model.Workspace{
+			ID:          workspaceID,
+			SignupToken: utils.CreateGUID(),
+		}
+
+		// insert
+		err := store.UpsertWorkspaceSignupToken(*workspace)
+		require.Nil(t, err)
```
Comment (wiggin77) [inline server/services/store/storetests/workspaces.go:50]: Minor thing... could you use `require.NoError` for these cases.  It adds some extra type checking for the test.

**Comment by yedamao** [comment]
@wiggin77 PTAL

File: server/services/store/storetests/workspaces.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		workspace := &model.Workspace{
+			ID: workspaceID,
+			Settings: map[string]interface{}{
+				"field1": "A",
+			},
+		}
+
+		// insert
+		err := store.UpsertWorkspaceSettings(*workspace)
+		// FIXME: insert workspace by store.UpsertWorkspaceSettings will failed
```
Comment (wiggin77) [inline server/services/store/storetests/workspaces.go:81]: @jespino is this expected?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: server/services/store/storetests/workspaces.go (hunk 2 above)
Comment (jespino) [reply server/services/store/storetests/workspaces.go:81]: No, it is not expected, you should be able to insert the workspace with the signup_token or make the signup_token not mandatory, clearly this is a bug.

Comment (jespino) [reply server/services/store/storetests/workspaces.go:81]: I think this should create a random signup_token on create, but keep it untouched on update.

**Comment by jespino** [comment]
I fixed the problem and merged the main branch. Thanks for finding this bug @yedamao! and for the new tests!

**Review by jespino — APPROVED** [review]

**Pull request merged**